### ISSUE TITLE
ProQuest: Not working on PDF pages

### ISSUE THREAD

**Issue by dstillman** [zotero/translators#1383]
https://forums.zotero.org/discussion/comment/282893/#Comment_282893

**Comment by zuphilip** [comment]
Probably also related to https://github.com/zotero/translators/issues/1148#issuecomment-248110524

**Comment by dstillman** [comment]
What's the connection with that other thread? As far as I can tell the detect code here is just looking for something (`open:block`) that no longer (?) exists on the PDF views. Is that not the case?

**Comment by zuphilip** [comment]
I haven't looked at the details, but remembered that we had similar reports for ProQuest databases before. It is a pity that they don't actively support the translator code anymore.

**Comment by zuphilip** [comment]
Okay, I looked now closer to this issue.

> As far as I can tell the detect code here is just looking for something (open:block) that no longer (?) exists on the PDF views. 

Yes, that seems correct. However, I don't see that the information about the item type is anywhere else in the pdf/fulltext page. The only solution I see is to open the abstract page and then proceed there...

**Comment by zuphilip** [comment]
Can we actually do a `doGet` in the `detectWeb`? I tried some code like this
```js
	Z.debug("NEW")
	var detailsUrl = ZU.xpathText(doc, '//li[@id="tab-AbstractRecord-null" or @id="tab-Record-null"]/a/@href')
	Z.debug(detailsUrl);
	if (detailsUrl) {
		ZU.doGet(detailsUrl, function(text, obj, url2) {
			parser = new DOMParser();
			text = parser.parseFromString(text, "text/html");
			var types = getTextValue(text, ["Source type", "Document type", "Record type"]);
			Z.debug(types);
			var zoteroType = getItemType(types);
			Z.debug(zoteroType);
			if (zoteroType) return zoteroType;
			//return detectWeb(obj, url2);
			
		});
	}
	
```
Everything seems fine, but in the end nothing is returned and detection fails.

**Comment by dstillman** [comment]
No, that won't work. `detectWeb()` needs a synchronous return value. What you have there will return `undefined`, with `return zoteroType` in the asynchronous callback.

**Comment by dstillman** [comment]
(We could probably make some changes to allow this, but generally we should avoid doing too much work on detection, so I'm not sure it's a good idea.)

**Comment by dstillman** [comment]
Does it work otherwise? I didn't look, but if that's the only issue it'd be better to set it to `document` as a quick fix and let it go on with the save, while we look into this further.

**Comment by zuphilip** [comment]
Yes, only the detection is not working on pdf/fulltext pages. I write a quick fix #1386 but uses `journalArticle` as fallback. Please have a look at this.
